cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -115,33 +115,32 @@ func printAvailableUpgrades(upgrades []upgrade.Upgrade, w io.Writer, featureGate
 			fmt.Fprintln(w, "")
 		}
 
-		fmt.Fprintf(w, "Upgrade to the latest %s:\n", upgrade.Description)
-		fmt.Fprintln(w, "")
-		fmt.Fprintln(tabw, "COMPONENT\tCURRENT\tAVAILABLE")
-		fmt.Fprintf(tabw, "API Server\t%s\t%s\n", upgrade.Before.KubeVersion, upgrade.After.KubeVersion)
-		fmt.Fprintf(tabw, "Controller Manager\t%s\t%s\n", upgrade.Before.KubeVersion, upgrade.After.KubeVersion)
-		fmt.Fprintf(tabw, "Scheduler\t%s\t%s\n", upgrade.Before.KubeVersion, upgrade.After.KubeVersion)
-		fmt.Fprintf(tabw, "Kube Proxy\t%s\t%s\n", upgrade.Before.KubeVersion, upgrade.After.KubeVersion)
-		if features.Enabled(featureGates, features.CoreDNS) {
-			fmt.Fprintf(tabw, "CoreDNS\t%s\t%s\n", upgrade.Before.DNSVersion, upgrade.After.DNSVersion)
+		if upgrade.Before.KubeadmVersion != upgrade.After.KubeadmVersion {
```

Comment:
The upgrade guides for [1.7->1.8](https://kubernetes.io/docs/tasks/administer-cluster/upgrade-downgrade/kubeadm-upgrade-1-8/) and [1.8->1.9](https://kubernetes.io/docs/tasks/administer-cluster/upgrade-downgrade/kubeadm-upgrade-1-9/) both recommend to use the _latest stable_ version of kubeadm for upgrades, i.e. not just the exact version you’d want to upgrade to. With that in mind this check doesn’t make sense; I would expect it to check if the kubeadm version is the same _or newer_.

#62568 already ensures that the major and minor versions are the same.

Reply:
Good feedback, thanks.

I'll need to resolve #62568 with what I'm trying to accomplish here. Generally, I don't want past versions to tell me what versions I need to upgrade to for things like KubeDNS because old versions have the wrong version numbers.